api/s2n.h

```diff
@@ -136,16 +136,48 @@ extern int *s2n_errno_location(void);
  *
  * This enum is optimized for use in C switch statements. Each value in the enum represents
  * an error "category".
+ *
+ * s2n-tls organizes errors into different "types" to allow applications to do logic on error
+ * values without catching all possibilities. Applications using non-blocking I/O should check
+ * error type to determine if the I/O operation failed because it would block or for some other
+ * error. To retrieve the type for a given error use `s2n_error_get_type()`. Applications should
+ * perform any error handling logic using these high level types:
+ *
+ * Here's an example that handles errors based on type:
+ *
+ * ```c
+ * s2n_errno = S2N_ERR_T_OK;
+ * if (s2n_recv(conn, &blocked) < 0) {
+ *     switch(s2n_error_get_type(s2n_errno)) {
+ *         case S2N_ERR_T_BLOCKED:
+ *             return -1;
+ *         case S2N_ERR_T_CLOSED:
+ *             return 0;
```

Comment:
I'm not sure how useful this example is. Some concerns:
* What does "return 0" mean vs "return -1"? Maybe define a SUCCESS vs FAILURE value.
* Shouldn't we retry on S2N_ERR_T_BLOCKED?
* Is S2N_ERR_T_CLOSED actually a "success" if you're trying to send data? The data probably wasn't sent.
* What would "handle_io_err" be vs "handle_proto_err"? Maybe just log them both.
* Your method signature for s2n_recv is wrong. Did you mean s2n_negotiate?

Reply:
This is a direct copy and paste from the Usage guide. I can follow up with a correction to the Usage guide after I have removed it from s2n.h